backend/src/graphql/resolver/TransactionResolver.ts

```diff
@@ -323,6 +323,7 @@ export class TransactionResolver {
     }
 
     // TODO this is subject to replay attacks
+    // --- WHY?
```

Comment:
This todo is still valid.

If you catch the request bob sends to the server and just send it again, the transaction is executed again.
We actually had this in our application when we didnt block the send button before executing the transaction. A very fast double klick had the transaction executed twice (tho this is not a replay attack).

A possible solution is to request a new transactionRequestToken, which can be used only once for a transaction.
If the transaction token is used, you cannot reuse it and have to request a new one to send another transaction.

This is overhead for very little benefit, but potential high security value. Therefore we skipped this in the past and decided against an implementation.

The semaphore does not solve this issue, since the server receives both calls and just waits (using the semaphore) before executing the second one - so this is not solved by semaphore(tho a semaphore for the transactionRequestToken would probably be needed)

Feel free to remove the `TODO` and make an issue for it instead.

Reply:
Remove the `Why?`